### PR TITLE
FR-NL-1.02 validation: element tree based approach

### PR DESCRIPTION
**Pull request by aaroncameron-wk**
#### Reason for change
The current line-by-line approach does not provide meaningful model object references in the output validation error. Move to a an element-tree based approach so useful refs can be included (while acknowledging that this approach isn't as thorough, it is more useful).

#### Description of change
Navigate through all elements and check all available text content for disallowed characters.

#### Steps to Test
CI

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/plugin/validate/NL/rules/br_kvk.py [hunk 1]
```diff
@@ -340,6 +341,8 @@ def rule_br_kvk_4_20(
     # Check period date values
     for contextId, context in modelXbrl.contexts.items():
         for contextElement in context:
+            if not isinstance(contextElement, ModelObject):
```
[the remark this answers is not in the document]
Comment (aaroncameron-wk) [reply arelle/plugin/validate/NL/rules/br_kvk.py:344]: `modelXbrl.contexts` is the collection of context elements, `contextElement` here is the direct descendants of that `context` element, whatever type of element they may be. Naming could probably be improved to `contextChild`.

**Review by derekgengenbacher-wf — APPROVED** [review]

File: arelle/plugin/validate/NL/rules/br_kvk.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (brettkail-wk) [reply arelle/plugin/validate/NL/rules/br_kvk.py]: 🤦 Of course, sorry for the noise.

**Review by austinmatherne-wk — APPROVED** [review]

**Comment by derekgengenbacher-wf** [comment]
QA +10 Testing locally succeeded.

**Pull request merged**